src/main/scala/org/broadinstitute/dsde/workbench/avram/Avram.scala

```diff
@@ -0,0 +1,43 @@
+package org.broadinstitute.dsde.workbench.avram
+
+import org.apache.commons.dbcp2.BasicDataSource
+import org.broadinstitute.dsde.workbench.avram.config.AvramConfig
+import org.broadinstitute.dsde.workbench.avram.util.SlickDatabaseFactory
+
+/**
+  * Object providing access to all services. This merges configuration and service code to provide
+  * one-stop access for endpoint implementations.
+  *
+  * Service clients:
+  * All code for which some form of dependency injection isn't possible should look here for
+  * services instead of manually reading configs and creating service instances.
+  *
+  * Service implementations:
+  * There are many possible techniques for providing access such as:
+  *   - a val that references a singleton instance (instance must be thread-safe)
+  *   - a def that creates new instances
+  *   - a loan function that wraps lifecycle management around a client-provided function
+  * The choice of technique depends on factors such as service thread-safety, instantiation cost,
+  * and resource limits.
+  *
+  * The technique should force (or at least encourage) correct usage. When feasible, it should be
+  * impossible for clients to misuse this access API. For example, while clients can't be stopped
+  * from holding on to references longer than they should, they can be encouraged to ask for new
+  * instances by providing a factory method that requires some appropriately scoped context data.
+  *
+  * The need for this is a consequence of Google Cloud Endpoints Framework controlling instance
+  * creation of the API implementation objects. However, introducing Guice
+  * (https://cloud.google.com/endpoints/docs/frameworks/java/using-guice) might give that control
+  * back to us and might be worth exploring.
+  */
+object Avram {
+  private val databaseFactory = new SlickDatabaseFactory(AvramConfig.dbcpDataSourceConfig)
+
+  val database = databaseFactory.database
```

Comment:
We may want to wrap the slick `database` in a class with utilities for using transactions, accessing data, etc. Right now Avram just runs raw sql queries but we probably want to lift that into a DB component layer.

For example Leo uses the cake pattern to break up database tables into different components, and provides a `DbReference` class for accessing the components within a transaction.

See: https://github.com/DataBiosphere/leonardo/blob/develop/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/DbReference.scala#L59

Usage:
```
dbRef.inTransaction { dataAccess =>
  dataAccess.clusterComponent.list()
}
```

I am wondering if it makes more sense to encapsulate `databaseFactory`, `database`, and `dbcpDataSource` in some sort of `DbReference` class, and have the `Avram` object just provide the `DbReference`.

Reply:
That sounds right to me. Anu is working on putting collections into the DB. I think we can apply the refactor you suggest here once we have something in the database to manage.

(I'll have to read up about the cake pattern again.)

Reply:
Yep, working on a PR to add Transactions and component layer!